### DISCUSSION TITLE
Substantial Samples Excluded from generate_beta_test_single()

### DISCUSSION THREAD

**Discussion by pkirti33** [cafferychen777/MicrobiomeStat#27] [Q&A] [answered]
Hello,
First and foremost, thank you for putting together this incredibly thorough and easy-to-use microbiome analysis package. It has allowed me to explore my data much more in-depth and even taught me more ways to explore it.

As far as I can tell, the format of my files matches the recommended format. My feature.tab/OTU table file is 96 columns (samples) by 9152 rows (probes in the microarray analysis). The only difference I've noticed is that my OTU table is binary (1 if a microbe is detected and 0 if it isn't), which differs from the examples in which there is a quantifiable amount of each bacteria in the sample. I've attempted to circumvent this issue by using the Jaccard method for beta diversity. 

My code is below. When I run it, the test appears to run without issues except I get a notice that "The following samples were excluded" and 80 of my samples are listed. Can someone please help me understand why so many of my samples are excluded from the analysis? I'm happy to provide more information if necessary.

```
generate_beta_test_single(data.obj = MicrobiomeData,
                          dist.obj = NULL,
                          time.var = "Trimester",
                          t.level = "2",
                          group.var = "Group",
                          adj.vars = NULL,
                          dist.name = c('Jaccard'))
```

**Comment by cafferychen777** [comment, marked as answer]
Hello @pkirti33,

Thank you for reaching out and for your kind words about the microbiome analysis package. I'm glad to hear that it's been helpful in your research.

Regarding your question about the exclusion of samples in your analysis, the key lies in the `t.level = "2"` parameter in your `generate_beta_test_single` function. This setting means that the function is configured to only include samples where the `Trimester` variable is equal to "2". Therefore, all samples that do not meet this criterion will be excluded from the analysis.

If your intention is to include more samples and not just those with `Trimester = "2"`, you might consider adjusting or removing the `t.level` parameter to broaden the scope of your analysis. Please let me know if you need further assistance or have more questions.

Best regards, 

Chen YANG
